src/apps/chat/drf/viewsets.py

```diff
@@ -1,27 +1,64 @@
 from core.drf.permissions import IsOwnerOrReadOnly
 from core.drf.serializers import ErrorResponseSerializer
 from django.shortcuts import get_object_or_404
+from django_filters.rest_framework import DjangoFilterBackend
 from drf_spectacular.utils import extend_schema
 from rest_framework import parsers, permissions, response, serializers, status
 from rest_framework.decorators import action
 from rest_framework.response import Response
 from rest_framework.viewsets import ModelViewSet
 
-from ..models import Message, MessageImage, Room, RoomUser
+from ..models import Message, MessageImage, Room, RoomUser, Category
 from .serializers import (
     MessageImageSerializer,
     MessageSeenSerializer,
     MessageSerializer,
     RoomSerializer,
     RoomUserSerializer,
     UserNameSerializer,
+    CategorySerializer,
 )
 
 
+# class CategoriesView(generics.ListCreateAPIView):
+#     queryset = Category.objects.all()
+#     serializer_class = CategorySerializer
+#     permission_classes = (permissions.IsAuthenticated,)
+#
+#     def perform_create(self, serializer):
+#         serializer.save(owner=self.request.user)
+#
+#     def get_queryset(self):
+#         return (
+#             super()
+#             .get_queryset()
+#             .filter(owner=self.request.user)
+#         )
+class CategoriesViewSet(ModelViewSet):
+    queryset = Category.objects.all()
+    serializer_class = CategorySerializer
+    permission_classes = (permissions.IsAuthenticated,
+                          IsOwnerOrReadOnly)
+    http_method_names = ['get', 'post', 'patch', 'delete']
+
+    def perform_create(self, serializer):
+        serializer.is_valid()
```

Comment:
No need to call `serializer.is_valid()` here. Check CreateModelMixin source code https://github.com/encode/django-rest-framework/blob/a25aac7d6739c43a4997829a0ff79cce12e8c0de/rest_framework/mixins.py#L25